### ISSUE TITLE
Detect leaking `for` loops in `ClassDef` and `Module`

### ISSUE THREAD

**Issue by sobolevn** [wemake-services/wemake-python-styleguide#2340]
Right now we can have two problems:

```python
class Some(object):
   for x in [1, 2]:
       print(x)

print(Some.x)  # oups, will show you `2`
```

and with modules:

```python
# some.py
for x in [1, 2]:
    print(x)

# __main__
import some
print(some.x)  # 2
```

This is not users want in 99.9% of cases. There are other - functional - solutions which don't leave extra vars in the scope.

Moreover, `for` is already forbidden in class bodies.

**Comment by alexeev-prog** [comment]
I can work with this

**Comment by alexeev-prog** [comment]
You need to check if there is a `for` loop inside the top (global) level of the module and inside the classes. To do this, you can add two warnings: one for the presence of for in the class and one for the presence of for in the global scope.

**Comment by sobolevn** [comment]
There's no need in two violations, one is good enough.

But, that's not all. We also need to define, when it is ok to use such code: 
1. `for x in ...: ...; del x`
2. `for x in ...: ...; globals().pop('x')`
3. What else?

`for _ in ...: ...` won't work, because `Klass._` is still not ideal.

**Comment by alexeev-prog** [comment]
We need check actions with objects? If we see del, globals.pop, any other action without creating - its ok?

**Comment by sobolevn** [comment]
Yes, because this way - name is not leaked.

**Comment by antriantsek** [comment]
i have a solution if you want to assign me..
